robot/impl/local_robot.go

```diff
@@ -1435,3 +1467,25 @@ func (r *localRobot) MachineStatus(ctx context.Context) (robot.MachineStatus, er
 func (r *localRobot) Version(ctx context.Context) (robot.VersionResponse, error) {
 	return robot.Version()
 }
+
+// checkMaintenanceSensorReadings ensures that errors from reading a sensor are handled properly.
+func (r *localRobot) checkMaintenanceSensorReadings(maintenanceAllowedKey string, sensor resource.Sensor) (bool, error) {
+	timeout := 5 * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Should this use the context passed into `Reconfigure` and not `context.Background()`?